web/src/main/java/com/zebrunner/reporting/web/TestCaseController.java

```diff
@@ -17,24 +20,28 @@
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestHeader;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 import javax.validation.Valid;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 @RequestMapping(path = "api/tests/cases", produces = MediaType.APPLICATION_JSON_VALUE)
 @RestController
 public class TestCaseController extends AbstractController implements TestCaseDocumentedController {
 
     private final Mapper mapper;
     private final TestCaseService testCaseService;
+    private final TestService testService;
     private final TestMetricService testMetricService;
 
-    public TestCaseController(Mapper mapper, TestCaseService testCaseService, TestMetricService testMetricService) {
+    public TestCaseController(Mapper mapper, TestCaseService testCaseService, TestService testService, TestMetricService testMetricService) {
         this.mapper = mapper;
         this.testCaseService = testCaseService;
+        this.testService = testService;
         this.testMetricService = testMetricService;
```

Comment:
Lombok's `@RequiredArgsConstructor` can gracefully eliminate this constructor, but its up to you.